tests/examples/shims/mouse.lua

```diff
@@ -17,11 +17,35 @@ function mouse.coords(args)
     return coords
 end
 
+function mouse.set_newindex_miss_handler(h)
+    rawset(mouse, "_ni_handler", h)
+end
+
+function mouse.set_index_miss_handler(h)
+    rawset(mouse, "_i_handler", h)
+end
+
 function mouse.push_history()
     table.insert(mouse.old_histories, mouse.history)
     mouse.history = {}
 end
 
-return mouse
+return setmetatable(mouse, {
+    __index = function(self, key)
+        if key == "screen" then
+            return screen[1]
+        end
```

Comment:
...why? There already is a `screen` property being set above, so this should be dead code (`__index` is not called if the requested member exists).
In case I am wrong:
...why not remove the (apparently non-working) `screen = screen[1]` above?

Reply:
Rebase issue...

The `if key == "screen" then ... end` is the right one. The other one breaks when you `fake_remove` the screen.